websites/S/Sonic Retro/presence.ts

```diff
@@ -0,0 +1,282 @@
+const presence = new Presence({
+		clientId: "970743721404530798"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			startTimestamp: browsingTimestamp
+		},
+		{ pathname, search, hostname, href } = window.location;
+	switch (hostname) {
+		case "sonicretro.org": {
+			presenceData.smallImageKey = "home_512";
+			if (search) {
+				presenceData.largeImageKey = "search";
+				if (!new URLSearchParams(search).get("s"))
+					presenceData.details = "Searching for something...";
+				else {
+					presenceData.details = `Searching for "${
+						document.querySelector("#main > div.archive-box > h1").textContent
+					}"`;
+				}
+			} else if (
+				!search &&
+				(pathname === "/" || pathname === "/#" || pathname.includes("/page/"))
+			) {
+				presenceData.largeImageKey = "home";
+				presenceData.details = "Zooming through the main page";
+			} else if (pathname.includes("/category/")) {
+				presenceData.largeImageKey = "search";
+				presenceData.details = `Looking through ${
+					document.querySelector("#main > div.archive-box > h1").textContent
+				}`;
+			} else if (
+				/https:\/\/sonicretro.org\/\d{4}\/\d{2}\/(\d{2})*\//.test(pathname)
+			) {
+				presenceData.largeImageKey = "search";
+				presenceData.details = `Looking for posts from ${
+					document.title.split("-")[0]
+				}`;
+			} else if (
+				/https:\/\/sonicretro.org\/([0-9]+(\/[0-9]+)+)\/+((\w|-|\d)*)\//g.test(
+					href
+				)
+			) {
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Reading an article";
+				presenceData.state = document.title.split(" - Sonic Retro")[0];
+				presenceData.buttons = [
+					{
+						label: "Read Article",
+						url: href
+					}
+				];
+			} else {
+				presenceData.largeImageKey = "logo_1024";
+				presenceData.details = "Viewing an unsupported page";
+			}
+			break;
+		}
+		case "info.sonicretro.org": {
+			// a full on wiki or index for the site
+			const unsupported = [
+				"Cite This Page - Sonic Retro",
+				"Compare pages - Sonic Retro",
+				"Export pages - Sonic Retro",
+				"What links here - Sonic Retro",
+				"Create account - Sonic Retro",
+				"Log in - Sonic Retro",
+				"Reset password - Sonic Retro",
+				"Login required - Sonic Retro",
+				"Preferences - Sonic Retro",
+				"Watchlist - Sonic Retro",
+				"Reset tokens - Sonic Retro",
+				"Blocked users - Sonic Retro",
+				"File list - Sonic Retro"
+			];
+			presenceData.smallImageKey = "info_512";
+			if (
+				pathname === "/" ||
+				pathname.includes("/Main_Page") ||
+				(pathname.includes("/index.php") && !search)
```

Comment:
did you really mean to put the !search check here or should it go on the entire if statement?

Reply:
No, just to that part.